operator/k8s_node.go

```diff
@@ -201,8 +231,9 @@ func runCNPNodeStatusGC(name string, clusterwide bool, ciliumNodeStore *store.Sh
 						needsUpdate := false
 						nodesToDelete := map[string]v1.Time{}
 						for n, status := range cnp.Status.Nodes {
-							kvStoreNodeName := nodeTypes.GetKeyNodeName(option.Config.ClusterName, n)
-							if _, exists := kvStoreNodes[kvStoreNodeName]; !exists {
+							nodeName := nodeTypes.GetKeyNodeName(option.Config.ClusterName, n)
```

Comment:
The keys are stored in a different format in k8s cache stores, IIRC it's simply the node name. This also means that this function will behave differently for kvstore GC (the current existing code).